src/main/java/at/hannibal2/skyhanni/features/misc/NoBitsWarning.kt

```diff
@@ -0,0 +1,31 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.BitsAPI
+import at.hannibal2.skyhanni.data.ProfileStorageData
+import at.hannibal2.skyhanni.events.BitsUpdateEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.SoundUtils
+import at.hannibal2.skyhanni.utils.SoundUtils.createSound
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.seconds
+
+class NoBitsWarning {
+
+    private val config get() = SkyHanniMod.feature.misc.noBitsWarning
+
+    private val profileStorage get() = ProfileStorageData.profileSpecific?.bits
+
+    @SubscribeEvent
+    fun onBitsUpdate(event: BitsUpdateEvent) {
+        if (!isEnabled()) return
+        if (BitsAPI.bitsToClaim == 0) {
+            ChatUtils.clickableChat("§bNo Bits Available! §eClick to run /bz booster cookie.", "bz booster cookie")
+            LorenzUtils.sendTitle("§bNo Bits Available", 5.seconds)
+            SoundUtils.repeatSound(100,10, createSound("note.pling", 0.6f))
+        }
+    }
+
+    private fun isEnabled() = LorenzUtils.inSkyBlock && profileStorage != null && config
```

Comment:
have you run into an issue where profile storage is null when calling the function because I dont see much reason for it to be null here. Maybe you could pass currentBits and bitsToClaim into the BitsUpdateEvent